### PR TITLE
update web template and move auth caching in base

### PR DESCRIPTION
**Pull request by aruniverse**

### CODE REVIEW

**Comment by aruniverse** [comment]
gah hold on might need itwin-client  😡
image

File: packages/modules/cra-template-web-viewer/template/src/App.tsx [hunk 1]
```diff
@@ -71,7 +45,7 @@ const App: React.FC = () => {
       } else {
         if (!process.env.IMJS_ITWIN_ID) {
           throw new Error(
-            "Please add a valid iTwin ID in the .env file and restart the application or add it to the iTwinId query parameter in the url and refresh the page. See the README for more information."
+            "Please add a valid context ID in the .env file and restart the application or add it to the iTwinId query parameter in the url and refresh the page. See the README for more information."
```
[the remark this answers is not in the document]
Comment (aruniverse) [reply packages/modules/cra-template-web-viewer/template/src/App.tsx:48]: gah, just copied and pasted the old 2.x template and move some pieces around will revert

File: packages/modules/cra-template-web-viewer/template/src/App.tsx [hunk 2]
```diff
@@ -89,27 +63,46 @@ const App: React.FC = () => {
   }, [isAuthorized]);
 
   useEffect(() => {
-    if (iTwinId && iModelId && isAuthorized) {
+    if (isAuthorized && iTwinId && iModelId) {
```
[the remark this answers is not in the document]
Comment (aruniverse) [reply packages/modules/cra-template-web-viewer/template/src/App.tsx:66]: just wanted to short circuit earlier in the if, can revert if youd like

**Comment by kckst8** [comment]
Added ViewerAuthorization class and exported for use; Removed the authClient from BaseInitializer

File: packages/modules/viewer-react/src/index.ts [hunk 3]
```diff
@@ -10,4 +10,4 @@ export * from "./utilities";
 export * from "./types";
 export * from "./hooks";
 export * from "oidc-client";
-export * from "./services/auth/ViewerAuthorizationClient";
+export * from "./services/auth";
```
Comment (aruniverse) [inline packages/modules/viewer-react/src/index.ts:13]: sweet!

making a comment here so we can revisit re-exporting the oidc-client pkg, and possiibly removing this altogether before next release

**Review by kckst8 — APPROVED** [review]

**Pull request merged**